opensoundscape/datasets.py

```diff
@@ -178,138 +179,174 @@ def collate_fn(*batch):
 
     Output:
         Dictionary:
-            { "X": (1, H, W) if spec_augment else (3, H, W)
+            { "X": (3, H, W)
             , "y": (1) if label_column != None
             }
     """
 
     def __init__(
         self,
         df,
-        audio_column="Destination",
+        filename_column="Destination",
+        from_audio=True,
         label_column=None,
         height=224,
         width=224,
         add_noise=False,
         debug=None,
-        spec_augment=False,
         random_trim_length=None,
-        overlay_prob = 0
+        overlay_prob=0,
+        overlay_weight="random",
     ):
         self.df = df
-        self.audio_column = audio_column
+        self.filename_column = filename_column
+        self.from_audio = from_audio
         self.label_column = label_column
         self.height = height
         self.width = width
         self.debug = debug
-        self.spec_augment = spec_augment
         self.random_trim_length = random_trim_length
         self.overlay_prob = overlay_prob
+        if (overlay_weight != "random") and (not 0 < overlay_weight < 1):
+            raise ValueError(
+                f"overlay_weight not in 0<overlay_weight<1 (given overlay_weight: {overlay_weight})"
+            )
+        self.overlay_weight = overlay_weight
+
+        self.transform = self.set_transform(add_noise=add_noise)
+        self.item_idx = None
+        self.row = None
 
+    def set_transform(self, add_noise):
+        transform_list = [transforms.Resize((self.height, self.width))]
         if add_noise:
-            self.transform = transforms.Compose(
+            transform_list.extend(
                 [
-                    transforms.Resize((self.height, self.width)),
                     transforms.RandomAffine(
                         degrees=0, translate=(0.2, 0.03), fillcolor=50
                     ),
                     transforms.ColorJitter(
                         brightness=0.3, contrast=0.3, saturation=0.3, hue=0
                     ),
-                    transforms.ToTensor(),
                 ]
             )
-        else:
-            self.transform = transforms.Compose(
-                [transforms.Resize((self.height, self.width)), transforms.ToTensor()]
+
+        transform_list.append(transforms.ToTensor())
+        return transforms.Compose(transform_list)
+
+    def random_audio_trim(self, audio, audio_length):
+        audio_length = len(audio.samples) / audio.sample_rate
+        if self.random_trim_length > audio_length:
+            raise ValueError(
+                f"the length of the original file ({audio_length} sec) was less than the length to extract ({self.random_trim_length} sec) for the file {audio_path}"
+            )
+        extra_time = audio_length - self.random_trim_length
+        start_time = np.random.uniform() * extra_time
+        trimmed = audio.trim(start_time, start_time + self.random_trim_length)
+        return trimmed, self.random_trim_length
```

Comment:
return of `self.audio_trim_length` is probably unnecessary here

Reply:
Fixed by https://github.com/kitzeslab/opensoundscape/pull/179/commits/bbfccac7e1e6eb9f73583f861c0720a2c1f45cc7